core/src/main/scala/io/github/gaelrenoux/tranzactio/ConnectionSource.scala

```diff
@@ -49,6 +66,13 @@ object ConnectionSource {
           }
       }
 
+    def runAutoCommitStream[R, E, A](task: Connection => ZStream[R, E, A])
+      (implicit errorStrategies: ErrorStrategiesRef, trace: Trace): ZStream[R, Either[DbException, E], A] =
+      ZStream.acquireReleaseWith(openConnection.mapError(Left(_)))(closeConnection(_).orDie).flatMap { (c: Connection) =>
+        ZStream.fromZIO(setAutoCommit(c, autoCommit = true).mapError(Left(_)))
+          .zipRight {task(c).mapError(Right(_))}
```

Comment:
Should be `crossRight` or we'll only keep the first element.

Reply:
I think executing the setAutoCommit to true is enough ones, it should stay the same for the whole connection

Reply:
Yes on the auto-commit sides, but it also only keeps one element on the task side (so only the first entry in the ZStream you've passed as a parameter). I've added some tests, and with `zipRight` it crashes because the only the first entry in the stream is actually found in the result.

Reply:
Okay then you are are right I thought it was merging the two stream in case of zipRight  as well just not executing the first before each element, only before start executing the stream. Then in this case what you think about a flatMap? I think it would be enough to execute the auto commit command ones.